rfcs/rfc-bn-messaging-queue.md

```diff
@@ -0,0 +1,268 @@
+		<CCRAHeader>
+			<userApplication>BI</userApplication>
+			<userRole>01</userRole>
+		</CCRAHeader>
+	</header>
+
+	<body>
+		<clientBasicInformationSearchInformationMessage>
+			<messageNumber>70377</messageNumber>
+			<messageEnglishText>Exact Match</messageEnglishText>
+			<messageFrenchText>Concordance exacte</messageFrenchText>
+		</clientBasicInformationSearchInformationMessage>
+		<clientBasicInformationSearchResult>
+			<businessRegistrationNumber>993775204</businessRegistrationNumber>
+			<legalName>Patrick Wei</legalName>
+			<operatingName>ROY&#39;S SOLE PROP</operatingName>
+			<businessTypeCode>01</businessTypeCode>
+			<corporationCertification>
+				<incorporationDate />
+				<incorporationCertificateIdentifier />
+				<jurisdictionCode />
+				<jurisdictionProvinceCode />
+				<jurisdictionCountryCode />
+				<jurisdictionStateCode />
+			</corporationCertification>
+			<businessAddress>
+				<foreignLegacy>
+					<addressDetailLine1>940 BLANSHARD ST</addressDetailLine1>
+					<addressDetailLine2 />
+				</foreignLegacy>
+				<municipality>VICTORIA</municipality>
+				<provinceStateCode>BC</provinceStateCode>
+				<postalCode>V8W3E6</postalCode>
+				<countryCode>CA</countryCode>
+				<returnMailIndicator>0</returnMailIndicator>
+			</businessAddress>
+			<mailingAddress>
+				<foreignLegacy>
+					<addressDetailLine1>940 BLANSHARD ST</addressDetailLine1>
+					<addressDetailLine2 />
+				</foreignLegacy>
+				<municipality>VICTORIA</municipality>
+				<provinceStateCode>BC</provinceStateCode>
+				<postalCode>V8W3E6</postalCode>
+				<countryCode>CA</countryCode>
+				<effectiveDate>2022-08-17</effectiveDate>
+				<returnMailIndicator>0</returnMailIndicator>
+				<careOfLine />
+			</mailingAddress>
+			<clientEvents>
+				<eventCode>03</eventCode>
+				<effectiveDate>2022-08-17</effectiveDate>
+				<expiryDate>9999-12-31</expiryDate>
+			</clientEvents>
+			<relatedBusinessRegistrationNumber />
+			<owners>
+				<ownerIndividualName>
+					<lastName>Wei</lastName>
+					<givenName>Patrick</givenName>
+				</ownerIndividualName>
+				<ownerBN />
+			</owners>
+		</clientBasicInformationSearchResult>
+	</body>
+</SBNClientBasicInformationSearchResponse>
+```
+
+# Approach
+
+Introduce a new queue entity-bn. This queue should handle both request based on the queue property. Queue should have a 
+business identifier/ business_id and request type (This will be used to identify which api call queue should trigger).
+Once registration filing is processed and business is created then entity-filer emits an event queue 
+(entity.business.bn), then entity-bn inform BC BN Hub about this new registration and once that is processed 
+entity-bn will emit an event queue (entity.business.bn) with request type (get-business-number) to itself, so that it
+can try to get the BN from OneStop (In case of failure it can try multiple times). This way we can track which call get
+failed and no need to send information to CRA multiple times.
```

Comment:
Typically what is the time frame with in which we decide that a call failed / how do we determine call failed? Should there a job that runs at predefined times to get the BN number?

Reply:
We get an error response from BC BN Hub (its not call timeout). I have considered the job in the 2nd option